src/radical/pilot/agent/scheduler/base.py

```diff
@@ -152,49 +270,49 @@ def _try_allocation(self, cu):
         CU off to the ExecutionWorker.
         """
 
-        # Get timestamp to use for recording a successful scheduling attempt
-        before_ts = time.time()
-
         # needs to be locked as we try to acquire slots, but slots are freed
         # in a different thread.  But we keep the lock duration short...
         with self._slot_lock :
 
+            self._prof.prof('schedule_try', uid=cu['uid'])
             # schedule this unit, and receive an opaque handle that has meaning to
             # the LRMS, Scheduler and LaunchMethod.
-            cu['opaque_slots'] = self._allocate_slot(cu['description']['cores'])
+            cu['slots'] = self._allocate_slot(cu['description'])
 
-        if not cu['opaque_slots']:
+        if not cu['slots']:
             # signal the CU remains unhandled
+            self._prof.prof('schedule_fail', uid=cu['uid'])
             return False
 
         # got an allocation, go off and launch the process
-        self._prof.prof('schedule', msg="try", uid=cu['uid'], timestamp=before_ts)
-        self._prof.prof('schedule', msg="allocated", uid=cu['uid'])
+        self._prof.prof('schedule_ok', uid=cu['uid'])
 
         if self._log.isEnabledFor(logging.DEBUG):
-            self._log.debug("slot status after allocated  : %s", self.slot_status())
-
-        self._log.debug("%s [%s] : %s [%s]", 
-                        cu['uid'], cu['description']['cores'], 
-                        cu['opaque_slots'], 
-                        len(cu['opaque_slots']['task_slots']))
+            self._log.debug("after  allocate   %s: %s", cu['uid'],
+                            self.slot_status())
 
+        self._log.debug("%s [%s/%s] : %s [%s]", cu['uid'],
+                        cu['description']['cpu_processes'],
+                        cu['description']['gpu_processes'],
+                        pprint.pformat(cu['slots']))
         return True
 
 
     # --------------------------------------------------------------------------
     #
     def reschedule_cb(self, topic, msg):
+
         # we ignore any passed CU.  In principle the cu info could be used to
         # determine which slots have been freed.  No need for that optimization
         # right now.  This will become interesting once reschedule becomes too
         # expensive.
+        # FIXME: optimization
 
         cu = msg
```

Comment:
Where is this needed? If it is only for passing the message in the next log, why not use it immediately?

Reply:
to make clear that the message is interpreted as a compute unit dict.  I try to keep naming of central variables consistent over different components...